### PR TITLE
Add probabilistic sampler processor

### PR DESCRIPTION
**Pull request by mar4uk**
#### PR Description
implements a flow component otelcol.processor.probabilistic_sampler that wraps over the upstream OpenTelemetry Collector Contrib probabilistic_sampler processor.

#### Which issue(s) this PR fixes

https://github.com/grafana/agent/issues/3596
#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+---
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.processor.probabilistic_sampler/
+labels:
+  stage: opensource
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:4]: Normally the value of "stage" will be one of the three values listed in the [Stability docs](https://grafana.com/docs/agent/latest/stability/):

- Experimental
- Beta
- Stable

`probabilistic_sampler` has different stabilities for logs (alpha) and traces (beta), so we should make that clear too.

Unfortunately, when I added the last few Otel Flow components, I forgot to set the stability labels on them. There is an ongoing discussion in these two PRs:
https://github.com/grafana/agent/pull/4948
https://github.com/grafana/agent/pull/4343

The main part is that we should not label something as less than "stable" if we have already listed it as "stable" in previous versions. So I'd suggest labeling `probabilistic_sampler` as `Experimental`, and we can change it later if it's more appropriate.

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+`otelcol.processor.probabilistic_sampler` is only reported as unhealthy if given an invalid
+configuration.
+
+## Debug information
+
+`otelcol.processor.probabilistic_sampler` does not expose any component-specific debug
+information.
+
+## Example
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:89]:
```suggestion
## Examples
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:90]:
```suggestion

### Basic usage
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  hash_seed           = 123
+  sampling_percentage = 15.3
+
+  output {
+    logs = [otelcol.exporter.otlp.default.input]
+  }
+}
+```
+
+Sample 15% of the logs:
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:102]:
```suggestion
### Sample 15% of the logs
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+otelcol.processor.probabilistic_sampler "default" {
+  sampling_percentage = 15
+
+  output {
+    logs = [otelcol.exporter.otlp.default.input]
+  }
+}
+```
+
+Sample logs according to their logID attribute:
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:114]:
```suggestion
### Sample logs according to their "logID" attribute
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+  attribute_source    = "record"
+  from_attribute      = "logID"
+
+  output {
+    logs = [otelcol.exporter.otlp.default.input]
+  }
+}
+```
+
+Sample logs according to the attribute `priority`:
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:128]:
```suggestion
### Sample logs according to a "priority" attribute
```

File: Makefile [hunk 7]
```diff
@@ -172,7 +172,7 @@ lint: agentlint
 # more without -race for packages that have known race detection issues.
 test:
 	$(GO_ENV) go test $(GO_FLAGS) -race ./...
-	$(GO_ENV) go test $(GO_FLAGS) ./pkg/integrations/node_exporter ./pkg/logs ./pkg/operator ./pkg/util/k8s ./component/otelcol/processor/tail_sampling ./component/loki/source/file
+	$(GO_ENV) go test $(GO_FLAGS) ./pkg/integrations/node_exporter ./pkg/logs ./pkg/operator ./pkg/util/k8s ./component/otelcol/processor/tail_sampling ./component/loki/source/file ./component/otelcol/processor/probabilistic_sampler
```
Comment (ptodev) [inline Makefile:175]: Is this change necessary? Does have probabilistic_sampler have race issues?

File: component/otelcol/processor/probabilistic_sampler/probabilistic_sampler.go [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+	Output *otelcol.ConsumerArguments `river:"output,block"`
+}
+
+var (
+	_ processor.Arguments = Arguments{}
+)
+
+// DefaultArguments holds default settings for Arguments.
+var DefaultArguments = Arguments{
+	AttributeSource: defaultAttributeSource,
```
Comment (ptodev) [inline component/otelcol/processor/probabilistic_sampler/probabilistic_sampler.go:49]: Since `defaultAttributeSource` is used only once, I think it might be better to just use it here.

File: component/otelcol/processor/probabilistic_sampler/probabilistic_sampler.go [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+			fact := psp.NewFactory()
+			return processor.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the otelcol.processor.probabilistic_sampler component.
+
+type Arguments struct {
+	SamplingPercentage float32 `river:"sampling_percentage,attr,optional"`
```
Comment (ptodev) [inline component/otelcol/processor/probabilistic_sampler/probabilistic_sampler.go:33]: I think `sampling_percentage` is not optional? The otel docs list it as required:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.80.0/processor/probabilisticsamplerprocessor/README.md

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+Trace ID hashing samples based on hash values determined by trace IDs.
+
+### Arguments
+
+For traces `otelcol.processor.probabilistic_sampler` supports the following arguments:
+
+Name | Type     | Description | Default | Required
+---- |----------| ----------- |---------| --------
+`hash_seed`               | `uint32` | An integer used to compute the hash algorithm. Note that all collectors for a given tier (e.g. behind the same load balancer) should have the same hash_seed. |         | no
+`sampling_percentage`     | `float32`| Percentage at which traces are sampled; >= 100 samples all traces | `0`     | no
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:47]: Same here - shouldn't `sampling_percentage` be required?

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+
+Name | Type      | Description | Default | Required
+---- |-----------| ----------- |---------| --------
+`hash_seed`               | `uint32`  | An integer used to compute the hash algorithm. Note that all collectors for a given tier (e.g. behind the same load balancer) should have the same hash_seed. |         | no
+`sampling_percentage`     | `float32` | Percentage at which traces are sampled; >= 100 samples all traces | `0`     | no
+`attribute_source`        | `string`  | Defines where to look for the attribute in from_attribute. The allowed values are `traceID` or `record` | `"traceID"`  | no
+`from_attribute`          | `string`  | The optional name of a log record attribute used for sampling purposes, such as a unique log record ID. The value of the attribute is only used if the trace ID is absent or if `attribute_source` is set to `record` | | no
+`sampling_priority`       | `string`  | The optional name of a log record attribute used to set a different sampling priority from the `sampling_percentage` setting. 0 means to never sample the log record, and >= 100 means to always sample the log record | | no
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:66]:
```suggestion
`sampling_priority`       | `string`  | The optional name of a log record attribute used to set a different sampling priority from the `sampling_percentage` setting. 0 means to never sample the log record, and >= 100 means to always sample the log record | `""` | no
```

If an argument is optional, we need to specify the default value.

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`input` | `otelcol.Consumer` | A value that other components can use to send telemetry data to.
+
+`input` accepts `otelcol.Consumer` data for any telemetry signal (metrics,
+logs, or traces).
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:77]:
```suggestion
`input` accepts `otelcol.Consumer` OTLP-formatted data for telemetry signals of these types:
* logs
* traces
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+### Arguments
+
+For logs `otelcol.processor.probabilistic_sampler` supports the following arguments:
+
+Name | Type      | Description | Default | Required
+---- |-----------| ----------- |---------| --------
+`hash_seed`               | `uint32`  | An integer used to compute the hash algorithm. Note that all collectors for a given tier (e.g. behind the same load balancer) should have the same hash_seed. |         | no
+`sampling_percentage`     | `float32` | Percentage at which traces are sampled; >= 100 samples all traces | `0`     | no
+`attribute_source`        | `string`  | Defines where to look for the attribute in from_attribute. The allowed values are `traceID` or `record` | `"traceID"`  | no
+`from_attribute`          | `string`  | The optional name of a log record attribute used for sampling purposes, such as a unique log record ID. The value of the attribute is only used if the trace ID is absent or if `attribute_source` is set to `record` | | no
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:65]: Ideally the "Description" field should be kept short. Then we would list the details under the table.

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+### Arguments
+
+For traces `otelcol.processor.probabilistic_sampler` supports the following arguments:
+
+Name | Type     | Description | Default | Required
+---- |----------| ----------- |---------| --------
+`hash_seed`               | `uint32` | An integer used to compute the hash algorithm. Note that all collectors for a given tier (e.g. behind the same load balancer) should have the same hash_seed. |         | no
+`sampling_percentage`     | `float32`| Percentage at which traces are sampled; >= 100 samples all traces | `0`     | no
+
+## Usage for logs
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:49]: All the Flow components have a consistent [page structure](https://github.com/grafana/agent/blob/main/docs/developer/writing-flow-component-documentation.md#page-structure). Would you mind reworking the docs to adhere to it please? It'd mean that we don't have a "Usage for logs" and "Usage for traces" sections. Instead, the "Arguments" section will need to explain how to use the component for traces and logs.

File: component/otelcol/processor/probabilistic_sampler/probabilistic_sampler_test.go [hunk 15]
```diff
@@ -0,0 +1,251 @@
...
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/pkg/flow/componenttest"
+	"github.com/grafana/dskit/backoff"
+	"github.com/grafana/river"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+func TestBadRiverConfigNegativeSamplingRate(t *testing.T) {
```
Comment (ptodev) [inline component/otelcol/processor/probabilistic_sampler/probabilistic_sampler_test.go:23]: It might be easier to have table driven tests like the ones in the [spanmetrics connector](https://github.com/grafana/agent/blob/main/component/otelcol/connector/servicegraph/servicegraph_test.go).

File: component/otelcol/processor/probabilistic_sampler/probabilistic_sampler_test.go [hunk 16]
```diff
@@ -0,0 +1,251 @@
...
+    attribute_source = "example"
+    output { 
+	    // no-op: will be overridden by test code.
+    }
+`
+	var args Arguments
+	require.EqualError(t, river.Unmarshal([]byte(exampleBadRiverConfig), &args), "invalid attribute source: example. Expected: traceID or record")
+}
+
+func TestLogProcessing(t *testing.T) {
```
Comment (ptodev) [inline component/otelcol/processor/probabilistic_sampler/probabilistic_sampler_test.go:46]: Those tests have a lot of boilerplate. It would be cleaner to do something like those ones:
https://github.com/grafana/agent/blob/main/component/otelcol/connector/spanlogs/spanlogs_test.go
https://github.com/grafana/agent/blob/main/component/otelcol/processor/attributes/attributes_test.go

**Review by ptodev — CHANGES_REQUESTED** [review]
Thank you for working on this! I added a few notes regarding docs and testing.

File: component/otelcol/processor/probabilistic_sampler/probabilistic_sampler.go (hunk 9 above)
Comment (mar4uk) [reply component/otelcol/processor/probabilistic_sampler/probabilistic_sampler.go:33]: yeah, it is listed as required, but actually, it isn't. If `sampling_percentage` is not specified then it is 0 by default. I will prepare PR upstream to fix README

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md (hunk 10 above)
Comment (mar4uk) [reply docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:47]: no, the default value is 0 = no sampling

File: Makefile (hunk 7 above)
Comment (mar4uk) [reply Makefile]: I thought it could be a race issue because tests failed a few times
Removed this change.
Tests are passing now. 
There is one test that failed a few times:
```
--- FAIL: Test_TopicDiscovery (1.00s)
    target_syncer_test.go:60: 
        	Error Trace:	/drone/src/component/loki/source/internal/kafkatarget/target_syncer_test.go:60
        	Error:      	Condition never satisfied
        	Test:       	Test_TopicDiscovery
        	Messages:   	expected topics: [topic1], got: []
```
I'm not sure if it is a flaky test or if it started to fail from time to time because of probabilistic_sampler...
https://drone.grafana.net/grafana/agent/12594/5/2

**Comment by mar4uk** [comment]
Thank you @ptodev for review! I fixed all of you mentioned

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 17]
```diff
@@ -0,0 +1,140 @@
+---
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.processor.probabilistic_sampler/
+labels:
+  stage: experimental
+title: otelcol.processor.probabilistic_sampler
+---
+
+# otelcol.processor.probabilistic_sampler
+
+`otelcol.processor.probabilistic_sampler` accepts logs and traces data from other otelcol components and apply probabilistic sampling based on configuration options.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:10]:
```suggestion
`otelcol.processor.probabilistic_sampler` accepts logs and traces data from other otelcol components and applies probabilistic sampling based on configuration options.
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+
+
+> **Note**: `otelcol.processor.probabilistic_sampler` is a wrapper over the upstream
+> OpenTelemetry Collector Contrib `probabilistic_sampler` processor. Bug reports or feature
+> requests will be redirected to the upstream repository, if necessary.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:15]:
```suggestion
{{% admonition type="note" %}}
`otelcol.processor.probabilistic_sampler` is a wrapper over the upstream
OpenTelemetry Collector Contrib `probabilistic_sampler` processor. If necessary, 
bug reports or feature requests will be redirected to the upstream repository.
{{% /admonition %}}
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 19]
```diff
@@ -0,0 +1,140 @@
...
+```
+
+## Arguments
+
+`otelcol.processor.probabilistic_sampler` supports the following arguments:
+
+Name | Type      | Description                                                                                                          | Default     | Required
+---- |-----------|----------------------------------------------------------------------------------------------------------------------|-------------| --------
+`hash_seed`               | `uint32`  | An integer used to compute the hash algorithm.                                                                       | `0`         | no
+`sampling_percentage`     | `float32` | Percentage at which traces or logs are sampled.                                                                      | `0`         | no
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:38]:
```suggestion
`sampling_percentage`     | `float32` | Percentage of traces or logs sampled.                                                                      | `0`         | no
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 20]
```diff
@@ -0,0 +1,140 @@
...
+`attribute_source`        | `string`  | Defines where to look for the attribute in `from_attribute`.                                                         | `"traceID"` | no
+`from_attribute`          | `string`  | The name of a log record attribute used for sampling purposes.                                                       | `""`        | no
+`sampling_priority`       | `string`  | The name of a log record attribute used to set a different sampling priority from the `sampling_percentage` setting. | `""`        | no
+
+`hash_seed` determines an integer to compute the hash algorithm. Could be used for both: traces and logs.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:43]:
```suggestion
`hash_seed` determines an integer to compute the hash algorithm. This argument could be used for both traces and logs.
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 21]
```diff
@@ -0,0 +1,140 @@
...
+When using for logs it computes the hash of a log record.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:44]:
```suggestion
When used for logs, it computes the hash of a log record.
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 22]
```diff
@@ -0,0 +1,140 @@
...
+In order for hashing to work, all collectors for a given tier (e.g. behind the same load balancer) must have the same `hash_seed`.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:45]:
```suggestion
For hashing to work, all collectors for a given tier, for example, behind the same load balancer, must have the same `hash_seed`. 
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 23]
```diff
@@ -0,0 +1,140 @@
...
+`attribute_source`        | `string`  | Defines where to look for the attribute in `from_attribute`.                                                         | `"traceID"` | no
+`from_attribute`          | `string`  | The name of a log record attribute used for sampling purposes.                                                       | `""`        | no
+`sampling_priority`       | `string`  | The name of a log record attribute used to set a different sampling priority from the `sampling_percentage` setting. | `""`        | no
+
+`hash_seed` determines an integer to compute the hash algorithm. Could be used for both: traces and logs.
+When using for logs it computes the hash of a log record.
+In order for hashing to work, all collectors for a given tier (e.g. behind the same load balancer) must have the same `hash_seed`. 
+It is also possible to leverage a different `hash_seed` at different collector tiers to support additional sampling requirements. 
+
+`sampling_percentage` determines the percentage at which traces or logs are sampled; >= 100 samples all
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:48]:
```suggestion
`sampling_percentage` determines the percentage at which traces or logs are sampled. All traces or logs are sampled if you set this argument to a value greater than or equal to 100.
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 24]
```diff
@@ -0,0 +1,140 @@
...
+
+`attribute_source` (logs only) determines where to look for the attribute in `from_attribute`. The allowed values are `traceID` or `record`.  
+
+`from_attribute` (logs only) determines the name of a log record attribute used for sampling purposes, such as a unique log record ID. The value of the attribute is only used if the trace ID is absent or if `attribute_source` is set to `record`
+
+`sampling_priority` (logs only) determines the name of a log record attribute used to set a different sampling priority from the `sampling_percentage` setting. 0 means to never sample the log record, and >= 100 means to always sample the log record
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:54]:
```suggestion
`sampling_priority` (logs only) determines the name of a log record attribute used to set a different sampling priority from the `sampling_percentage` setting. 0 means to never sample the log record, and greater than or equal to 100 means to always sample the log record.
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 25]
```diff
@@ -0,0 +1,140 @@
...
+`hash_seed` determines an integer to compute the hash algorithm. Could be used for both: traces and logs.
+When using for logs it computes the hash of a log record.
+In order for hashing to work, all collectors for a given tier (e.g. behind the same load balancer) must have the same `hash_seed`. 
+It is also possible to leverage a different `hash_seed` at different collector tiers to support additional sampling requirements. 
+
+`sampling_percentage` determines the percentage at which traces or logs are sampled; >= 100 samples all
+
+`attribute_source` (logs only) determines where to look for the attribute in `from_attribute`. The allowed values are `traceID` or `record`.  
+
+`from_attribute` (logs only) determines the name of a log record attribute used for sampling purposes, such as a unique log record ID. The value of the attribute is only used if the trace ID is absent or if `attribute_source` is set to `record`
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:52]:
```suggestion
`from_attribute` (logs only) determines the name of a log record attribute used for sampling purposes, such as a unique log record ID. The value of the attribute is only used if the trace ID is absent or if `attribute_source` is set to `record`.
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 26]
```diff
@@ -0,0 +1,140 @@
...
+
+`sampling_priority` (logs only) determines the name of a log record attribute used to set a different sampling priority from the `sampling_percentage` setting. 0 means to never sample the log record, and >= 100 means to always sample the log record
+
+The `probabilistic_sampler` supports two types of sampling for traces:
+1. `sampling.priority` [semantic
+   convention](https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table) as defined by OpenTracing
+2. Trace ID hashing
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:59]:
```suggestion
1. `sampling.priority` [semantic
   convention](https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table) as defined by OpenTracing.
2. Trace ID hashing.
```

**Review by clayton-cornell — COMMENTED** [review]
Some doc suggestions

**Comment by mar4uk** [comment]
Thank you @clayton-cornell! I made changes to address all your suggestions

File: component/otelcol/processor/probabilistic_sampler/probabilistic_sampler.go [hunk 27]
```diff
@@ -0,0 +1,87 @@
...
+	FromAttribute      string  `river:"from_attribute,attr,optional"`
+	SamplingPriority   string  `river:"sampling_priority,attr,optional"`
+
+	// Output configures where to send processed data. Required.
+	Output *otelcol.ConsumerArguments `river:"output,block"`
+}
+
+var (
+	_ processor.Arguments = Arguments{}
+)
```
Comment (ptodev) [inline component/otelcol/processor/probabilistic_sampler/probabilistic_sampler.go:41]:
```suggestion
	_ river.Validator = (*Arguments)(nil)
	_ river.Defaulter = (*Arguments)(nil)
)
```

File: docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md [hunk 28]
```diff
@@ -0,0 +1,142 @@
...
+labels:
+  stage: experimental
+title: otelcol.processor.probabilistic_sampler
+---
+
+# otelcol.processor.probabilistic_sampler
+
+`otelcol.processor.probabilistic_sampler` accepts logs and traces data from other otelcol components and applies probabilistic sampling based on configuration options.
+
+
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.probabilistic_sampler.md:12]:
```suggestion
```

**Review by ptodev — APPROVED** [review]
Thank you, it looks great! I added a few minor suggestions. After @clayton-cornell approves, we could merge it.

**Comment by ptodev** [comment]
Seems there are a few [linter errors](https://drone.grafana.net/grafana/agent/12663/1/2) in the CI

**Comment by mar4uk** [comment]
forgot to import river, now linter passes

**Review by clayton-cornell — APPROVED** [review]

**Comment by clayton-cornell** [comment]
@ptodev @mar4uk I've approved for docs... looks like a minor conflict needs a poke and then it should be done.
